velox/exec/StreamingWindow.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/exec/Operator.h"
+#include "velox/exec/RowContainer.h"
+#include "velox/exec/Window.h"
+#include "velox/exec/WindowFunction.h"
+#include "velox/exec/WindowPartition.h"
+
+namespace facebook::velox::exec {
+
+/// This operator differs from the Window operator in that it does not require
+/// sorting the input vector or caching all the input data. Instead, it groups
+/// the input batch, performs calculations on first n-1 partitions, and carries
+/// forward the last partition to the next input batch.
+class StreamingWindow : public Window {
+ public:
+  StreamingWindow(
+      int32_t operatorId,
+      DriverCtx* driverCtx,
+      const std::shared_ptr<const core::WindowNode>& windowNode)
+      : Window(operatorId, driverCtx, windowNode) {}
```

Comment:
I'm not sure it is a good idea to use Window operator as a base class. This makes it so that operatorName is the same between these two, which is confusing. Consider removing the inheritance and share code some other way.

I'm also seeing that pretty much all methods and state of Window operator are now protected and available to SteamingWindow. This feels excessive. Would be nice to tighten access.

Reply:
@mbasmanova : I too feel similarly about this. 

In Presto based window itself there are implementations for partially ordered windows (not implemented though).

Do we really need a separate StreamingWindow implementation ? It is using a lot of underlying structures and assumptions of Window. Can the logic not be incorporated into the current Window code ?

Also, this implementation requires each partition to fit into memory. That might not be valid. And some sort of Spill to disk would be required.

Reply:
@JkSelf : Please can you open a GitHub issue/discussion about this topic. We can finalize a decision there.

Reply:
@aditi-pandit Open https://github.com/facebookincubator/velox/discussions/5925 for StreamingWindow.